Mathlib/Analysis/SpecialFunctions/Log/Poslog.lean

```diff
@@ -0,0 +1,157 @@
+standard estimates.
+-/
+
+open Real
+
+/-!
+## Definition, Notation and Reformulations
+-/
+/-- Definition: the real part of the logarithm. -/
+noncomputable def Real.poslog : ℝ → ℝ := fun r ↦ max 0 (log r)
+
+/-- Notation `log⁺` for the real part of the logarithm. -/
+notation "log⁺" => poslog
+
+/-- Definition of the real part of the logarithm, formulated as a theorem. -/
+theorem Real.poslog_def {r : ℝ} : log⁺ r = max 0 (log r) := rfl
+
+/-!
+## Elementary Properties
+-/
+/-- Presentation of `log` in terms of its positive part. -/
+theorem Real.log_eq_poslog_sub_poslog_inv {r : ℝ} : log r = log⁺ r - log⁺ r⁻¹ := by
+  rw [poslog_def, poslog_def, log_inv]
+  by_cases h : 0 ≤ log r
+  · simp [h]
+  · rw [not_le] at h
+    simp [neg_nonneg.1 (Left.nonneg_neg_iff.2 h.le)]
+
+/-- Presentation of `log⁺` in terms of `log`. -/
+theorem Real.poslog_eq_half_mul_log_add_log_abs {r : ℝ} : log⁺ r = 2⁻¹ * (log r + |log r|) := by
+  by_cases hr : 0 ≤ log r
+  · simp [poslog, hr, abs_of_nonneg]
+    ring
+  · simp [poslog, le_of_not_ge hr, abs_of_nonpos]
+
+/-- The positive part of `log` is never negative. -/
+theorem Real.poslog_nonneg {x : ℝ} : 0 ≤ log⁺ x := by simp [poslog]
+
+/-- The function `log⁺` is even. -/
+theorem Real.poslog_neg (x : ℝ) : log⁺ x = log⁺ (-x) := by simp [poslog]
+
+/-- The function `log⁺` is even. -/
+theorem Real.poslog_abs (x : ℝ) : log⁺ |x| = log⁺ x := by simp [poslog]
+
+/-- The function `log⁺` is zero in the interval [-1,1]. -/
+theorem Real.poslog_eq_zero_iff (x : ℝ) : log⁺ x = 0 ↔ |x| ≤ 1 := by
+  rw [← poslog_abs, ← log_nonpos_iff (abs_nonneg x)]
+  simp [poslog]
+
+/-- The function `log⁺` equals `log` outside of in the interval (-1,1). -/
+theorem Real.poslog_eq_log {x : ℝ} (hx : 1 ≤ |x|) : log⁺ x = log x := by
+  simp [poslog]
+  rw [← log_abs]
+  apply log_nonneg hx
+
+/-- The function `log⁺` equals `log` for all natural numbers. -/
+theorem Real.poslog_eq_log_of_nat {n : ℕ} : log n = log⁺ n := by
+  by_cases hn : n = 0
+  · simp [hn, poslog]
+  · simp [poslog_eq_log, Nat.one_le_iff_ne_zero.2 hn]
+
+/-- The function `log⁺` is monotone in the positive axis. -/
+theorem Real.monotoneOn_poslog : MonotoneOn log⁺ (Set.Ici 0) := by
+  intro x hx y hy hxy
+  simp [poslog]
+  by_cases h : log x ≤ 0
+  · tauto
+  · right
+    have := log_le_log (lt_trans Real.zero_lt_one ((log_pos_iff hx).1 (not_le.1 h))) hxy
+    simp [this]
+    linarith
+
+/-!
+## Estimates for Products
+-/
+/-- Estimate for `log⁺` of a product. See `Real.poslog_prod` for a variant involving
+multiple factors. -/
+theorem Real.poslog_mul {a b : ℝ} :
+    log⁺ (a * b) ≤ log⁺ a + log⁺ b := by
+  by_cases ha : a = 0
+  · simp [ha, poslog]
+  by_cases hb : b = 0
+  · simp [hb, poslog]
+  unfold poslog
+  nth_rw 1 [← add_zero 0, log_mul ha hb]
+  exact max_add_add_le_max_add_max
+
+/-- Estimate for `log⁺` of a product. Special case of `Real.poslog_mul` where one of
+the factors is a natural number. -/
+theorem Real.poslog_mul_nat {n : ℕ} {a : ℝ} :
```

Comment:
`posLog_nat_mul`, as the natural comes to the left of the multiplication.

Reply:
done.